src/linear_algebra/finsupp.lean

```diff
@@ -632,3 +633,18 @@ lemma mem_span_finset {s : finset M} {x : M} :
     (show x ∈ span R (id '' (↑s : set M)), by rwa set.image_id) in
   ⟨v, hvx ▸ (finsupp.total_apply_of_mem_supported _ hvs).symm⟩,
 λ ⟨f, hf⟩, hf ▸ sum_mem _ (λ i hi, smul_mem _ _ $ subset_span hi)⟩
+
+universe u
+
+/-- An element `m ∈ M` is contained in the `R`-submodule spanned by a set `s ⊆ M`, if and only if
+`m` can be written as a finite `R`-linear combination of elements of `s`.
+The implementation uses `finsupp.sum`. -/
+lemma mem_span_set {m : M} {s : set M} :
+  m ∈ submodule.span R s ↔
+  ∃ c : M →₀ R, (c.support : set M) ⊆ s ∧ c.sum (λ mi r, r • mi) = m :=
+begin
+  convert @finsupp.mem_span_iff_total _ M R _ _ _ id s m,
```

Comment:
Any reason for the `@` and `_`? It seemed to work for me without.

Reply:
It did, but, when possible, I prefer to see non-vanishing goals: without the @, there were 4 goals after the first line, which then resolved to only 1 after the second line.  As it is now, there are two goals after the `convert` and they are the ones solved in the brackets below.

Mostly it is aesthetic...

Reply:
Perhaps then it would be better to use:
```suggestion
  conv_lhs { rw ←set.image_id s },
  convert finsupp.mem_span_iff_total R,
```
(and remove the line below) so that the vanishing goals never appear, but the underscores are still avoided.

Reply:
Or overall:
```lean
begin
  conv_lhs { rw ←set.image_id s },
  simp_rw ←exists_prop,
  exact finsupp.mem_span_iff_total R,
end
```

Reply:
Interesting! I think the general preference is to keep the code clean; extra goals which disappear automatically don't contribute too much to the noise in the tactic state, do they? Maybe worth gathering opinions on Zulip if you feel strongly about it.

Reply:
I do not feel too strongly about it, but between the proof above and the one below, I do prefer the shorter one that also has no ghost goals.

```
begin
  convert finsupp.mem_span_iff_total R,
  exact (set.image_id s).symm,
  ext,
  split; exact λ ⟨hsub, hsum⟩, ⟨hsub, hsum⟩,
end
```